### PR TITLE
OmniTool first pass

### PR DESCRIPTION
**Pull request by jherico**
This is a new library that will allow for modular behavior using hand controllers based on information stored in entity userdata.

Full documentation pending implementation of some modules.

### CODE REVIEW

**Comment by ZappoMan** [comment]
Why did you rename these constants.js and utilities.js to austinConstants.js, etc??? Seems like we should keep the original names.

File: examples/libraries/omniTool.js [hunk 1]
```diff
@@ -0,0 +1,304 @@
+//
+//  Created by Bradley Austin Davis on 2015/09/01
+//  Copyright 2015 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+Script.include("austinConstants.js");
```
Comment (ZappoMan) [inline examples/libraries/omniTool.js]: why have these labeled "austinXXX.js"?

Comment (jherico) [inline examples/libraries/omniTool.js]: I didn't feel confident about putting something called constants or utils into the main libraries, but I'll update.

File: examples/libraries/omniTool.js [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+//
+
+Script.include("constants.js");
+Script.include("utils.js");
+Script.include("highlighter.js");
+Script.include("omniTool/models/modelBase.js");
+Script.include("omniTool/models/wand.js");
+
+// FIXME brad's requests
+// callback for unloading
```
Comment (ZappoMan) [inline examples/libraries/omniTool.js]: looks like you have this in the latest code... you can remove this fixme comment

**Pull request merged**